models/repo/user_repo.go

```diff
@@ -170,3 +170,26 @@ func GetReviewers(ctx context.Context, repo *Repository, doerID, posterID int64)
 	users := make([]*user_model.User, 0, 8)
 	return users, db.GetEngine(ctx).Where(cond).OrderBy(user_model.GetOrderByName()).Find(&users)
 }
+
+// GetPullRequestPosters returns all users that have authored a pull request for the given repository
+func GetPullRequestPosters(ctx context.Context, repo *Repository) ([]*user_model.User, error) {
+	e := db.GetEngine(ctx)
+	userIDs := make([]int64, 0, 8)
+	if err := e.Table("issue").
+		Cols("poster_id").
+		Where("is_pull = ?", true).
+		And("repo_id = ?", repo.ID).
+		Distinct("poster_id").
+		Find(&userIDs); err != nil {
+		return nil, err
+	}
+
+	users := make([]*user_model.User, 0, len(userIDs))
```

Comment:
You can do this in one query ...

Reply:
If you need help to craft the join just ask

Reply:
Thanks for the tip! I use a different ORM in my projects, so wasn't familiar with how to do some things in xorm.

Let me know if f6d774a9da442cbbf77e61cca6165515551de16a looks ok 🙂